src/client-private.ts

```diff
@@ -41,26 +50,104 @@ function setupStreamingClient (this: PureCloudWebrtcSdk): Promise<void> {
-      log.call(this, 'info', 'PureCloud streaming client connected', { reconnect: this._hasConnected });
-      this._hasConnected = true;
-      // refresh turn servers every 6 hours
-      this._refreshTurnServersInterval = setInterval(this._refreshTurnServers.bind(this), 6 * 60 * 60 * 1000);
-      return connection.webrtcSessions.refreshIceServers();
-    })
-    .then(e => {
-      log.call(this, 'info', 'PureCloud streaming client ready for WebRTC calls');
-    });
+
+  await connection.connect();
+  this.emit('connected', { reconnect: this._hasConnected });
+  log.call(this, 'info', 'PureCloud streaming client connected', { reconnect: this._hasConnected });
+  this._hasConnected = true;
+  // refresh turn servers every 6 hours
+  this._refreshTurnServersInterval = setInterval(this._refreshTurnServers.bind(this), 6 * 60 * 60 * 1000);
+  await connection.webrtcSessions.refreshIceServers();
+  log.call(this, 'info', 'PureCloud streaming client ready for WebRTC calls');
 }
 
-function startMedia (this: PureCloudWebrtcSdk) {
+export function proxyStreamingClientEvents (this: PureCloudWebrtcSdk) {
+  // webrtc events
+  const on = this._streamingConnection.webrtcSessions.on.bind(this._streamingConnection);
+  on('requestIncomingRtcSession', onPendingSession.bind(this));
+  on('incomingRtcSession', onSession.bind(this));
+  on('rtcSessionError', this.emit.bind(this, 'error'));
+  on('cancelIncomingRtcSession', session => this.emit('cancelPendingSession', session));
+  on('handledIncomingRtcSession', session => this.emit('handledPendingSession', session));
+  on('traceRtcSession', this.emit.bind(this, 'trace'));
+
+  // other events
+  this._streamingConnection.on('error', this.emit.bind(this, 'error'));
+  this._streamingConnection.on('disconnected', () => this.emit('disconnected', 'Streaming API connection disconnected'));
+}
+
+export async function startGuestScreenShare (this: PureCloudWebrtcSdk): Promise<void> {
+  const stream = await startDisplayMedia.call(this);
+  const jid = parseJwt(this._jwt).data.jid;
+  this._streamingConnection.webrtcSessions.initiateRtcSession({
+    stream,
+    jid,
+    mediaPurpose: 'screenshare'
+  });
+  // Should we attach the video stream on the guest's DOM?
+  // attachScreenShareMedia.call(this, stream);
+  temporaryOutboundStream = stream;
+}
+
+function getDefaultChromeConstraints (): MediaStreamConstraints {
+  if (window.navigator.mediaDevices.getDisplayMedia) {
+    return {
+      audio: false,
+      video: {
+        displaySurface: 'monitor'
+      }
+    } as MediaStreamConstraints;
+  }
+  return {
+    audio: false,
+    video: {
+      mandatory: {
+        chromeMediaSource: 'desktop',
+        maxWidth: window.screen.width,
+        maxHeight: window.screen.height,
+        maxFrameRate: 15
+      }
+    }
+  } as MediaStreamConstraints;
+}
+
+function startDisplayMedia (this: PureCloudWebrtcSdk): Promise<MediaStream> {
+  if (window.navigator.getDisplayMedia) {
```

Comment:
This might not be necessary. I can't think of why we'd be using this. It's never existed that I know of - maybe in desktop app for something?